sound/soc/sof/trace.c

```diff
@@ -53,73 +50,64 @@ static int sof_wait_trace_avail(struct snd_sof_dev *sdev, size_t *count,
 
 	if (signal_pending(current)) {
 		remove_wait_queue(&sdev->trace_sleep, &wait);
-		goto _endcheck;
+		goto out;
 	}
 
 	/* set timeout to max value, no error code */
 	schedule_timeout(MAX_SCHEDULE_TIMEOUT);
 	remove_wait_queue(&sdev->trace_sleep, &wait);
 
-_endcheck:
-	/* calculate the available count */
-	avail = sdev->host_offset - pos;
-
-_host_end:
-	/* return min value between available and request count */
-	*count = avail < *count ? avail : *count;
-
-	return 0;
+out:
+	/* return bytes available for copy */
+	if (sdev->host_offset < pos)
+		return buffer_size - pos;
+	else
+		return sdev->host_offset - pos;
```

Comment:
the first goto out could actually go straight to the last line, no need to retest twice if (sdev->host_offset < pos)